### PR TITLE
Add CI jobs for new and deprecated metrics

### LINKED ISSUE #1494: [CI] Add ci jobs without `ENABLE_METRICS_PREVIEW` (bazel) and `WITH_METRICS_PREVIEW` (cmake)
**Is your feature request related to a problem?**

I notice the new metrics implement is enabled only when building without `ENABLE_METRICS_PREVIEW` (bazel) and `WITH_METRICS_PREVIEW` (cmake). But it's not checked in most CI jobs. I think we should add some steps or jobs to build without them.

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1494 (issue)

## Changes

makes all jobs testing the new metrics and adds `bazel` and `cmake`, <del>`gcc 4.8`</del> jobs for testing deprecated metrics.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: ci/do_ci.sh [hunk 1]
```diff
@@ -242,6 +277,10 @@ elif [[ "$1" == "bazel.test" ]]; then
   bazel $BAZEL_STARTUP_OPTIONS build $BAZEL_OPTIONS //...
```
Comment (owent) [inline ci/do_ci.sh:277]: Should `BAZEL_OPTIONS="$BAZEL_OPTIONS --copt=-DENABLE_METRICS_PREVIEW"` above be removed?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,6 +23,22 @@ jobs:
         sudo ./ci/setup_thrift.sh
         ./ci/do_ci.sh cmake.test
 
+  cmake_deprecated_metrics_test:
+    name: CMake test (without otlp-exporter and deprecated metrics)
```
Comment (lalitb) [inline .github/workflows/ci.yml:27]:
```suggestion
    name: CMake test (without otlp-exporter and with deprecated metrics)
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -165,7 +181,27 @@ jobs:
         sudo ./ci/install_bazelisk.sh
     - name: run tests
       run: ./ci/do_ci.sh bazel.test
-
+  bazel_deprecated_metrics_test:
```
Comment (lalitb) [inline .github/workflows/ci.yml:184]: nit - add newline before and after this test.

**Review by lalitb — APPROVED** [review]
LGTM - few nit comments. Thanks for fixing the memory issues.

**Review by owent — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (esigo) [reply .github/workflows/ci.yml:185]: thanks, done

**Pull request merged**